From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: burakgon <burakgon12@gmail.com>
Date: Mon, 28 Sep 2026 20:08:41 +0300
Subject: [PATCH 12/12] cta: avoid out-of-bounds array access

Port of upstream cb5e3ed ("vic: avoid out-of-bounds array access") to the
0.3.0 layout: VIC 220 indexed _di_cta_video_formats[220], one past the
end of the 220-entry table.

(cherry picked from commit cb5e3edcec6a3f653aab297976b0ebbf8ad5a0d5)
---
 cta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cta.c b/cta.c
index b78d898..b0af09d 100644
--- a/cta.c
+++ b/cta.c
@@ -38,7 +38,7 @@
 const struct di_cta_video_format *
 di_cta_video_format_from_vic(uint8_t vic)
 {
-	if (vic > _di_cta_video_formats_len ||
+	if (vic >= _di_cta_video_formats_len ||
 	    _di_cta_video_formats[vic].vic == 0)
 		return NULL;
 	return &_di_cta_video_formats[vic];
